test/controllers/unattended_controller_test.rb

```diff
@@ -291,208 +470,40 @@ class UnattendedControllerTest < ActionController::TestCase
-      SETTINGS[:organizations_enabled] = true
+      ptable_ubuntu = FactoryBot.create(:ptable, :ubuntu, :name => 'ubuntu default',
+        :layout => 'd-i partman-auto/disk string /dev/sda\nd-i partman-auto/method string regular...',
+                                         :operatingsystem_ids => [operatingsystems(:ubuntu1010).id])
+      @host_with_template_subnet = FactoryBot.create(:host, :managed, :with_dhcp_orchestration, :with_templates_subnet, :build => true,
+                              :operatingsystem => operatingsystems(:ubuntu1010),
+                              :ptable => ptable_ubuntu,
+                              :medium => media(:ubuntu),
+                              :architecture => architectures(:x86_64)
+                             )
     end
 
-    test "hosts with mismatched ip and update_ip=false should have the old ip" do
-      disable_orchestration # avoids dns errors
-      Setting[:token_duration] = 30
-      Setting[:update_ip_from_built_request] = false
+    test "hosts with a template proxy which supplies a templateServer should use it" do
+      template_server_from_proxy = 'https://someproxy:8443'
+      ProxyAPI::Template.any_instance.stubs(:template_url).returns(template_server_from_proxy)
       @request.env["REMOTE_ADDR"] = '127.0.0.1'
-      h=@ub_host
-      h.create_token(:value => "aaaaaa", :expires => Time.now.utc + 5.minutes)
-      get :built, params: { 'token' => h.token.value }
-      h_new=Host.find_by_name(h.name)
-
-      assert_response :success
-      assert_equal h.ip, h_new.ip
+      @host_with_template_subnet.create_token(:value => "aaaaad", :expires => Time.now.utc + 5.minutes)
+      get :host_template, params: { :kind => 'provision', 'token' => @host_with_template_subnet.token.value }
+      assert @response.body.include?("#{template_server_from_proxy}/unattended/finish?token=aaaaad")
     end
 
-    test "hosts with mismatched ip and update_ip true should have the new ip" do
-      disable_orchestration # avoids dns errors
+    test "hosts with a template proxy with no templateServer should use the proxy name" do
       Setting[:token_duration] = 30
-      Setting[:update_ip_from_built_request] = true
-      h=@ub_host
-      new_ip = h.subnet.network.gsub(/\.0$/, '.100') # Must be in the subnet, which isn't fixed
-      @request.env["REMOTE_ADDR"] = new_ip
-      refute_equal new_ip, h.ip
-      h.create_token(:value => "aaaaab", :expires => Time.now.utc + 5.minutes)
-      get :built, params: { 'token' => h.token.value }
-      h_new=Host.find_by_name(h.reload.name)
-      assert_response :success
-      assert_equal new_ip, h_new.ip
-    end
-
-    test "hosts with mismatched ip and update_ip true and a duplicate ip should succeed with no ip update" do
-      disable_orchestration # avoids dns errors
-      Setting[:token_duration] = 30
-      Setting[:update_ip_from_built_request] = true
-      @request.env["REMOTE_ADDR"] = @rh_host.ip
```

Comment:
This is the only test using both hosts, but it doesn't really matter which existing host this is AFAICT so i replaced this with just a new host.